### PR TITLE
Rework FX output

### PR DESCRIPTION
**Pull request by josesimoes**
- Rework DebugPrint in FX output.
  (.NET nanoFramework doesn't implement String.Enumerator and the CLR now throws a not supported exception when called)
- Update several .NET nanoFramework NuGets.

### CODE REVIEW

**Comment by josesimoes** [comment]
@xinchen10 the build for the _other_  frameworks is failing. You probably have an idea of what could be wrong... 😉

File: amqp.sln [hunk 1]
```diff
@@ -132,15 +132,15 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Amqp.NetMF44", "netmf\Amqp.
...
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Amqp.Net40", "csproj\Amqp.Net40.csproj", "{4A250B81-35E9-4D2F-8030-62909F6C8C63}"
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Amqp.Net40", .csproj\Amqp.Net40.csproj", "{4A250B81-35E9-4D2F-8030-62909F6C8C63}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Amqp.NetCore", "csproj\Amqp.NetCore.csproj", "{35F64888-3446-444D-8E44-128378434AB3}"
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Amqp.NetCore", .csproj\Amqp.NetCore.csproj", "{35F64888-3446-444D-8E44-128378434AB3}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Amqp.Uwp", "csproj\Amqp.Uwp.csproj", "{75DC3484-F6C8-4373-A8C4-9D1A8C9FC3B0}"
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Amqp.Uwp", .csproj\Amqp.Uwp.csproj", "{75DC3484-F6C8-4373-A8C4-9D1A8C9FC3B0}"
 EndProject
```
Comment (xinchen10) [inline amqp.sln:144]: Build failed due to changes in this file. I looks like the changes were unintentional.

**Comment by josesimoes** [comment]
Oooops... Indeed.

File: amqp.sln (hunk 1 above)
Comment (josesimoes) [reply amqp.sln]: Should be reverted now. Seems that our automated update script needs a fix...

**Comment by josesimoes** [comment]
Now there there is a test failing. Not sure if it's again something else inadvertently changed...

**Pull request merged**